frame/society/src/mock.rs

```diff
@@ -142,7 +142,7 @@ impl EnvBuilder {
 		}
 	}
 
-	pub fn execute<R, F: FnOnce() -> R>(mut self, f: F) -> R {
+	pub fn execute(mut self, f: impl FnOnce() -> ()) {
```

Comment:
why?

Reply:
thanks for spotting this! previously had intentions of adding the `try_state` hook here but it seems it's not required for `pallet-society` ?

Reply:
the tips pallet having simple check, thought adding it to this pr would be sufficient, i'll keep it to just one pallet going forward